badius-osic-green-tribes/src/test/java/com/greenfox/exam/badiusosicgreentribes/transaction/handler/ProductionHandlerTest.java

```diff
@@ -0,0 +1,140 @@
+        Storage storage = Storage.builder()
+                .gold(100)
+                .food(50)
+                .build();
+        kingdom.setStorage(storage);
+        Production transaction = Production.builder()
+                .productionType(ProductionUnitType.GOLD)
+                .quantity(10)
+                .targetKingdom(kingdom)
+                .state(TransactionState.COMPLETED)
+                .build();
+
+        handler.confirm(transaction);
+
+        assertEquals(110, storage.getGold());
+        assertEquals(50, storage.getFood());
+    }
+
+    @Test
+    void confirm_NotCompletedTransaction() {
+        Kingdom kingdom = new Kingdom();
+        Storage storage = Storage.builder()
+                .gold(100)
+                .food(50)
+                .build();
+        kingdom.setStorage(storage);
+        Production transaction = Production.builder()
+                .productionType(ProductionUnitType.GOLD)
+                .quantity(10)
+                .targetKingdom(kingdom)
+                .state(TransactionState.CANCELLED)
+                .build();
+
+        handler.confirm(transaction);
+
+        assertEquals(100, storage.getGold());
+        assertEquals(50, storage.getFood());
+    }
+
+    @Test
+    void confirm_CompletedTransaction_Farm() {
+        Kingdom kingdom = new Kingdom();
+        Storage storage = new Storage();
+        storage.setBuildings(new ArrayList<>());
+        kingdom.setStorage(storage);
+        Production transaction = Production.builder()
+                .productionType(ProductionUnitType.FARM)
+                .quantity(1)
+                .targetKingdom(kingdom)
+                .state(TransactionState.COMPLETED)
+                .build();
+
+        handler.confirm(transaction);
+
+        assertTrue(kingdom.getStorage().getBuildings().stream()
+                .anyMatch(building -> building.getType() == BuildingType.FARM));
+    }
+
+    @Test
+    void confirm_CompletedTransaction_Knight() {
+        Kingdom kingdom = new Kingdom();
+        Storage storage = new Storage();
+        kingdom.setStorage(storage);
+
+        Army defenderArmy = new Army();
+        List<Troop> troops = new ArrayList<>();
+
+        Troop newTroop = new Troop();
+        Unit unit = new Unit();
+        unit.setType(UnitType.KNIGHT);
+        newTroop.setUnit(unit);
+        troops.add(newTroop);
+        defenderArmy.setTroops(troops);
+
+        storage.setDefenderArmy(defenderArmy);
+
+        Production transaction = Production.builder()
+                .productionType(ProductionUnitType.KNIGHT)
+                .quantity(1)
+                .targetKingdom(kingdom)
+                .state(TransactionState.COMPLETED)
+                .build();
+
+        handler.confirm(transaction);
+
+        assertTrue(kingdom.getStorage().getDefenderArmy().getTroops().stream()
+                .anyMatch(troop -> troop.getUnit().getType() == UnitType.KNIGHT));
+    }
```

Comment:
Tranzakció state ide sem kell. 

Ez is paraméterezhető, hogy le tudd vele fedni minden egység gyártását. 
Argument lista: 

1. ProductionUnitType ( csak a unit-ok)